src/memefactory/ui/get_dank/events.cljs

```diff
@@ -28,6 +28,35 @@
    {:db (assoc db
           ::spinner false)}))
 
+(re-frame/reg-event-fx
+ ::stage
+ (fn [{:keys [db]} [_ stage-number]]
+   {:db (assoc db :memefactory.ui.get-dank.page/stage stage-number)}))
+
+(defn get-allocated-dank [db phone-number]
+  (js/Promise.
+   (fn [resolve reject]
+     (web3-eth/contract-call
+      (contract-queries/instance db :dank-faucet)
+      :allocated-dank
+      (web3/sha3 phone-number)
+      (fn [error data]
+        (if error
+          (reject error)
+          (resolve (first (aget data "c")))))))))
+
+(re-frame/reg-event-fx
+ ::check-for-preallocated-dank
+ (fn [{:keys [db]} [_ {:keys [country-code phone-number] :as data}]]
+   (-> (get-allocated-dank db phone-number)
```

Comment:
Correct way would be to make `get-allocated-dank` into a reframe effect, because re-frame events should stay pure. 

https://github.com/Day8/re-frame/blob/master/docs/Effects.md

Reply:
@madvas Please check the commit below.